### PR TITLE
Fix ctrl-c causing segfault

### PR DESCRIPTION
**Pull request by JackDunnNZ**
Ref #52. This seems to work fine on Linux, but is patchy on OS X (I just had it work while running an unrelated solve, but most of the time, ctrl-c still causes a segfault). Not sure if you want to merge, but here it is anyway.

@joehuchette Does the CPLEX version have the same OS X problems, or is that just happening here?

### CODE REVIEW

**Comment by joehuchette** [comment]
I haven't seen any weird behavior on the current implementation, but I definitely noticed unpredictable behavior on OSX on some previous attempts that seemed like they should've worked fine. Debugging it was beyond my pay grade, though.

File: src/grb_common.jl [hunk 1]
```diff
@@ -53,6 +53,29 @@ macro grb_ccall(func, args...)
     end
 end
 
+macro grb_ccall_intercept(model, func, args...)
+    f = "GRB$(func)"
+    quote
+        ccall(:jl_exit_on_sigint, Void, (Cint,), convert(Cint,0))
+        ret = try
+            $(@windows? :(ccall(($f,libgurobi), stdcall, $(args...))) : :(ccall(($f,libgurobi), $(args...))) )
```
Comment (mlubin) [inline src/grb_common.jl]: Have you tried this on windows?

Comment (JackDunnNZ) [inline src/grb_common.jl]: Just tried it, it still solves normally. Sending ctrl-c inside cmd or powershell does nothing, and inside cygwin it segfaults.

Gurobi and JuMP tests are both fine FWIW

**Comment by mlubin** [comment]
Just to clarify, what does gurobi do on ctrl-c on linux before this patch? Does it segfault?

**Comment by JackDunnNZ** [comment]
Good point. I just tried it and it throws an InterruptException, but the
solve seems to keep running in the background according to CPU usage. With
the patch, it catches the exception, calls terminate, then rethrows, but
calling terminate doesn't seem to stop the solve either. I'm not sure
what's actually going on...

**Comment by mlubin** [comment]
So with this patch the solver keeps running in the background?

**Comment by JackDunnNZ** [comment]
Yes, it does, so as is this changes nothing. It would work if `terminate`
was changed to the appropriate command to actually stop the solver.

**Comment by mlubin** [comment]
Maybe the call to ``terminate`` needs to happen inside of a callback?

**Comment by JackDunnNZ** [comment]
It's possible that this is the problem, `terminate` is listed with the other callback methods, although the others explicitly say they can only be used during specific callbacks, while `terminate` says it can be called any time during the optimization (but maybe that just means regardless of the state during the callback). 

If that's the case, how do we get around it? My first guess would be to set a flag in the model to mark terminated and then set up an info callback to call `terminate` if that flag is set. I presume we can't add a new callback during a solve, but if we could then we could just add one that calls `terminate`.Any thoughts?

**Comment by mlubin** [comment]
Probably worth asking on the gurobi google group to see what they suggest (and to ask why ``terminate`` doesn't work).

**Comment by rdeits** [comment]
@JackDunnNZ did you ever hear back from the gurobi group about this? Being able to interrupt the solver would be extremely useful.

**Comment by JackDunnNZ** [comment]
Sorry, I never pursued this further

**Pull request closed without merging**